src-java/ping-topology/ping-storm-topology/src/main/java/org/openkilda/wfm/topology/ping/bolt/FlowFetcher.java

```diff
@@ -427,7 +428,17 @@ private void emitCacheExpire(Tuple input, CommandContext commandContext, Set<Flo
         OutputCollector collector = getOutput();
         flowsSet.removeAll(flows);
         for (FlowWithTransitEncapsulation flow : flowsSet) {
-            Values output = new Values(flow.getFlow(), commandContext);
+            CacheExpirationRequest cacheExpirationRequest;
+            if (flow.getHaFlow() != null) {
+                cacheExpirationRequest = new CacheExpirationRequest(flow.getHaFlow().getHaFlowId(),
```

Comment:
I think the code will be more readable if you add either 
`CacheExpirationRequest(FlowWithTransitEncapsulation flow)`
or `FlowWithTransitEncapsulation(Flow flow)` + `FlowWithTransitEncapsulation(HaFlow haFlow)` constructors.
This would let you avoid copy-pasting `flow.getXFlow()` calls several times.
As an option, `FlowWithTransitEncapsultation.toCacheExpirationRequest()` can play the trick as well.

Reply:
done